tests/intel_fpga_test.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# Executes a bunch of small test for the Intel FPGA backend
+# These are not intended to be performance test: they just check that everything compiles
+# and the correct result is produced
+
+set -a
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+PYTHONPATH=$SCRIPTPATH/..
+
+DACE_debugprint="${DACE_debugprint:-0}"
+ERRORS=0
+FAILED_TESTS=""
+TESTS=0
+
+TEST_TIMEOUT=10
+
+RED='\033[0;31m'
+YELLOW='\033[0;33m'
+NC='\033[0m'
+
+
+################################################
+
+bail() {
+    ERRORSTR=$1
+    /bin/echo -e "${RED}ERROR${NC} in $ERRORSTR" 1>&2
+    ERRORS=`expr $ERRORS + 1`
+    FAILED_TESTS="${FAILED_TESTS} $ERRORSTR\n"
+}
+
+run_sample() {
+    # Args:
+    #  1 - Relative path of FPGA test starting from test folder
+    #  2 - Name of the DAPP program
+    #  3 - a string indicating the list of input to pass to the python program (the transformation sequence)
+    #  4 - program command line argument (if any)
+
+    TESTS=`expr $TESTS + 1`
+    echo -e "${YELLOW}Running test $1...${NC}"
+
+    #1: generate the opencl
+    # This will throw an exception, because the kernel has not yet been built. Catch this, and build the kernel.
+    echo -e ${3} | python3 ${1}.py ${@:4} 2> /dev/null|:
+
+    #2: compile for emulation
+    cd .dacecache/$2/build
+    make intelfpga_compile_$2_emulator
+    if [ $? -ne 0 ]; then
+      bail "$1 (${RED}high-level synthesis failed${NC})"
+      return 1
+    fi
+
+    #3: execute the emulation
+    cd ../../../
+    echo -e $3 | python3 $1.py ${@:4}
+
+    if [ $? -ne 0 ]; then
+        bail "$1 (${RED}Wrong emulation result${NC})"
+    fi
+
+    #4 cleanup
+    cd .dacecache/$2/build
+    rm -fr $1_*
+    cd -
+
+
+
+    return 0
+}
+
+run_all() {
+    # #### VECTORIZATION ####
+    #Vectorization 1: first vectorize and then transform for FPGA
+    run_sample intel_fpga/vec_sum vec_sum "11\n1\n"
```

Comment:
Does this use the numbered CLI interface? That will not work in the long term, as the numbers will change whenever transformations are modified or added. We have to use Python to select the transformation of the correct type.

Reply:
Good observation! I've fixed this, thanks